e2e/imgbuild/imgbuild.go

```diff
@@ -1589,44 +1589,80 @@ func (c *imgBuildTests) testContainerBuildUnderFakerootModes(t *testing.T) {
 	)
 }
 
+func (c *imgBuildTests) testSIFHeaderAndExecute(t *testing.T) {
+	tmpDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "test-SIF-header-fields-", "")
+	defer cleanup(t)
+
+	path := fmt.Sprintf("%s/img.sif", tmpDir)
+
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs(path, "docker://busybox"),
+		e2e.ExpectExit(0),
+	)
+
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("sif"),
+		e2e.WithArgs("header", path),
+		e2e.ExpectExit(0,
+			e2e.ExpectOutput(e2e.ContainMatch, "#!/usr/bin/env run-singularity"),
+			e2e.ExpectOutput(e2e.RegexMatch, ".*[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}.*"),
+		))
+
+	output, err := exec.Command(path, "echo", "hello").Output()
+	if err != nil {
+		t.Errorf("could not execute the sif image: %s, error: %s", path, err.Error())
+		t.Fail()
+	}
+	if string(output) != "hello\n" {
+		t.Errorf("the output is not expected, expected: hello\n, actual: %s", string(output))
```

Comment:
I think it's better to not include `\n` in the output message, it will just introduce a new line.

Reply:
okay, I've added `-n` to ignore the newline.

Reply:
not sure about the reason, but code shown here is not changed, so please check the `Files Changed` tab to see the latest updates.

Reply:
I am looking at Files changed and still see the `\n` in the string, even after a reload.  I wasn't thinking of adding `-n`, I thought it would be fine for the comparison to include the newline, I just didn't think it would be good in the printed message.

Reply:
ohh, really sorry for the misunderstood, anyway, I removed newline in both the echo and output message.